core/src/main/java/org/hisp/dhis/android/core/program/internal/ProgramIndicatorHandler.kt

```diff
@@ -29,23 +29,42 @@ package org.hisp.dhis.android.core.program.internal
 
 import dagger.Reusable
 import javax.inject.Inject
+import org.hisp.dhis.android.core.arch.db.querybuilders.internal.WhereClauseBuilder
 import org.hisp.dhis.android.core.arch.db.stores.internal.IdentifiableObjectStore
 import org.hisp.dhis.android.core.arch.handlers.internal.HandleAction
 import org.hisp.dhis.android.core.arch.handlers.internal.IdentifiableHandlerImpl
 import org.hisp.dhis.android.core.arch.handlers.internal.LinkHandler
 import org.hisp.dhis.android.core.arch.handlers.internal.OrderedLinkHandler
+import org.hisp.dhis.android.core.common.IdentifiableColumns
 import org.hisp.dhis.android.core.common.ObjectWithUid
 import org.hisp.dhis.android.core.legendset.ProgramIndicatorLegendSetLink
 import org.hisp.dhis.android.core.program.AnalyticsPeriodBoundary
 import org.hisp.dhis.android.core.program.ProgramIndicator
 
 @Reusable
 internal class ProgramIndicatorHandler @Inject constructor(
-    programIndicatorStore: IdentifiableObjectStore<ProgramIndicator>,
+    private val programIndicatorStore: IdentifiableObjectStore<ProgramIndicator>,
     private val programIndicatorLegendSetLinkHandler: OrderedLinkHandler<ObjectWithUid, ProgramIndicatorLegendSetLink>,
     private val analyticsPeriodBoundaryHandler: LinkHandler<AnalyticsPeriodBoundary, AnalyticsPeriodBoundary>
 ) : IdentifiableHandlerImpl<ProgramIndicator>(programIndicatorStore) {
 
+    override fun afterCollectionHandled(oCollection: Collection<ProgramIndicator>?) {
+        val inDbProgramIndicatorUids = programIndicatorStore.selectUids()
+        val apiProgramIndicatorUids = oCollection?.map(ProgramIndicator::uid)
+        val deleteProgramIndicatorUid = inDbProgramIndicatorUids.filter { inDbProgramIndicatorUid ->
+            val isPresentOnline = apiProgramIndicatorUids?.contains(inDbProgramIndicatorUid)
+            isPresentOnline == false
+        }
+        val query = WhereClauseBuilder().apply {
+            deleteProgramIndicatorUid.forEach { uid ->
```

Comment:
There is a method `appendInKeyStringValues` that uses the IN operator and avoids the iteration of the array. But you should that `deleteProgramIndicatorUid` is not empty in first place